src/auth/auth.service.ts

```
@@ -0,0 +1,67 @@
+import { AuthDto } from './dto/AuthDto';
+import { ConfigService } from '@nestjs/config';
+import { NotFoundException } from '@nestjs/common/exceptions';
+import { OAuth2Client } from 'google-auth-library';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { JwtService } from '@nestjs/jwt';
+
+const client = new OAuth2Client(
+  process.env.GOOGLE_CLIENT_ID,
+  process.env.GOOGLE_CLIENT_SECRET,
+);
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prismaService: PrismaService,
+    private config: ConfigService,
+    private jwt: JwtService,
+  ) {}
+
+  async signUp(oAuthToken: string) {
+    try {
+      const { email, name } = await this.googleAuth(oAuthToken);
+      const userDto = new AuthDto({ email, name });
+      const userExist = await this.prismaService.user.findUnique({
+        where: { email },
+      });
+      if (!userExist) {
+        const user = await this.prismaService.user.create({
+          data: {
+            ...userDto,
+          },
+        });
+
+        return this.signToken(user.id);
+      }
+      return this.signToken(userExist.id);
```

If the email is already taken we want to raise an exception.
According to this code, you can bypass the login.